### DISCUSSION TITLE
re-rendering issue

### DISCUSSION THREAD

**Discussion by karthic-keyan** [d3/d3#3699] [Q&A]
whenever the component is re-render i am getting like this (see image)
Framework -> react
code:

> const simulation = d3
    .forceSimulation()
    .nodes(nodeData)
    .force("charge", d3.forceManyBody().strength(-400)) // repulsion btw the node, -ve repulse +ve attract each other
    .force("center", d3.forceCenter(width / 2, height / 2)) // center
    .force(
      "collide",
      d3
        .forceCollide()
        .radius(90)
        .strength(90)
        .iterations(1)
    );
  if (nodeData?.length) {
    simulation
      .force(
        "link",
        d3
          .forceLink(edgeData)
          .id(link => link.id)
          .distance(300) // distance btw the node & link
      )
      .on("tick", tickFn);
  }

**Comment by Fil** [comment]
Try and make sure that the code in _tickFn_ doesn't append a node each time it is run, but instead updates the existing nodes.

**Reply by karthic-keyan** [reply]
hey the problem is -> i am having the same data, wheneve i click something the graph component will get re-render so how can i avoid this

**Reply by Fil** [reply]
There is no way I can help you with so little information.
